vfs/vfs_storage.go

```diff
@@ -540,3 +598,53 @@ func (d *Driver) GetBackupInfo(backupURL string) (map[string]string, error) {
 func (d *Driver) ListBackup(destURL string, opts map[string]string) (map[string]map[string]string, error) {
 	return objectstore.List(opts[OPT_VOLUME_NAME], destURL, d.Name())
 }
+
+func flock(volume *Volume) (uintptr, error) {
+	cf, err := volume.ConfigFile()
+	if err != nil {
+		return 0, err
+	}
+	lockPath := cf + ".lock"
+
+	lockChan := make(chan struct {
+		ptr uintptr
+		err error
+	})
+	go func() {
+		file, err := os.Create(lockPath)
+		if err != nil {
+			lockChan <- struct {
+				ptr uintptr
+				err error
+			}{0, err}
+		}
+		err = syscall.Flock(int(file.Fd()), syscall.LOCK_EX)
+		lockChan <- struct {
+			ptr uintptr
+			err error
+		}{file.Fd(), err}
+	}()
+
+	select {
+	case result := <-lockChan:
+		return result.ptr, result.err
+	case <-time.After(time.Second * 10):
```

Comment:
This timeout logic doesn't seem good because if you timeout the background flock is still running and will acquire the lock eventually and then hold up everything.  I would just not timeout.  It seems like there is no good way to do this because I don't know who you would interrupt the syscall.  You put logic that if it timed out and the flock eventually succeeds, it immediately unlocks.  But that seems racy.